### ISSUE TITLE
missing TOC on \subpage

### ISSUE THREAD

**Issue by reekes** [doxygen/doxygen#10097]
**Describe the bug**

Doxygen is no longer producing a Table of Contents on pages referenced as /subpage

**To Reproduce**

1. Create a page "Subpage" which has \tableofcontents and a few sections.
2. Include "Subpage" using \subpage.

Example included below

**Version**
Mac 1.9.7

**Stack trace**
no crash occurs

**Additional context**
This was working in previous MacOS versions, including 1.9.5, 1.9.6, and all other versions I've used. I have not tried the 1.9.7 release on Windows or some Linux variation

In previous versions, there would be an HTML section near the top of the page containing the TOC block. The HTML headers (e.g., H1, H2, ...) have a tag `id=autotoc_md6`. This is still happening in 1.9.7 as it was in previous versions.

In previous versions, there's a section within the HTML `div class="toc"` which appeared just after `div class="contents"`. This is no longer produced by (Mac) v1.9.7

```
/**
\file
\mainpage Demonstrate TOC Bug
\tableofcontents

\note Table of Contents does appear on this page

# Header 1
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

## Header 1.1
Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.

# Header 2
Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

# Test TOC on other pages

- \ref ref-page "Page shows TOC"
- \subpage sub-page "Page does not show TOC"


\page ref-page TOC does appear on Ref Page
\tableofcontents

# Header 1
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

## Header 1.1
Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.

# Header 2
Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

\page sub-page TOC does not appear on Sub Page
\tableofcontents

# Header 1
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

## Header 1.1
Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.

# Header 2
Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
*/
```

**Comment by albert-github** [comment]
Checked it against the executables I have and the last one conforming the 1.9.6 version was
```
Commit: 7277a9a5156c4e8f370ae1d97615c82b23a822e1 [7277a9a]
Date: Wednesday, April 19, 2023 9:22:46 PM
Merge pull request #9981 from albert-github/feature/bug_latex_warnings_2

Warnings in LaTeX output (2)
```
the first different one was:
```
Commit: 5189f59d67cc3095674385aca8063a03045be89d [5189f59]
Date: Thursday, April 20, 2023 7:54:49 PM
Merge branch 'mtc-mlx-mtc/latex-group-hierarchy'
```

Note the current master (1.9.8 (0ff318b3cef7cd379fe2660ea0a58a7868103008)) still has the bad behavior.

**Comment by doxygen** [comment]
@reekes Please verify if the referenced commit fixes the problem for you. Do not close the issue, as this will be done automatically at the next official release.

**Comment by reekes** [comment]
@doxygen sorry but I don't have a way to build this. I manage software projects, and write our documentation, but I don't have the build tools. I can write code, but too many other tasks as a manager. If you provide a binary I can test it.

**Comment by albert-github** [comment]
@reekes Which OS?
see for the latest build executables https://github.com/doxygen/doxygen/actions/runs/5111710200 (includes this fix)

**Comment by reekes** [comment]
Yes!
[macOS Latest Release build artifacts](https://github.com/doxygen/doxygen/suites/13217215556/artifacts/719973413) fixes this.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
